src/common/components/markdownViewer/syncScroll/synchronizeScroll.ts

```diff
@@ -0,0 +1,113 @@
+
+const SOURCE_LINE_CLASSNAME = 'sourceLine';
+const SOURCE_LINE_ATTRIBUTE = 'line';
+const SOURCE_LINE_HIGHLIGHT_CLASSNAME = 'highlightSourceLine';
+
+interface Candidate {
+  lineNum: number;
+  element: Element;
+}
+
+interface CandidateSearch {
+  exactMatch: boolean;
+  candidate: Candidate;
+  nextCandidate?: Candidate;
+}
+
+const isIntegerProperty = (property: string) => {
+  return Number.isInteger(Number(property));
+};
+
+// Retrieve the candidate(s) elements that correspond to a certain
+// source line for an input list of rendered elements.
+const getRenderedElementsForSourceLine = (elementList: NodeListOf<Element>, targetLine: number): CandidateSearch => {
+  let candidate: Candidate = null;
+  if (elementList[0] && isIntegerProperty(elementList[0].getAttribute(SOURCE_LINE_ATTRIBUTE))) {
+    candidate = {
+      lineNum: Number(elementList[0].getAttribute(SOURCE_LINE_ATTRIBUTE)),
+      element: elementList[0],
+    };
+  }
+
+  for (const element of elementList) {
+    const slAttr = element.getAttribute(SOURCE_LINE_ATTRIBUTE);
+    if (isIntegerProperty(slAttr)) {
+      const current: Candidate = { lineNum: Number(slAttr), element };
+      if (current.lineNum === targetLine) {
+        return { exactMatch: true, candidate: current };
+      }
+      else if (current.lineNum > targetLine) {
+        return { exactMatch: false, candidate, nextCandidate: current };
+      }
+      candidate = current;
+    }
+  }
+  return { exactMatch: false, candidate };
+};
```

Comment:
I think we coudl apply clean coding to this function (same level of abstraction level and single responsibility principle).

This method is hard to read, I see two main steps

GetTheInitialCandidate

The go looping through all the elemtns in the list, looking form something and go updating candidate?

Can you provide a summary of what this method makes and we can make a refactor proposal.

Reply:
@fjcalzado could you provide psuedocode describing what this method does so we can refactoring applying clean code?

Reply:
I made a refactor trying to simplify complexity and break the code into smaller parts with basic responsability and shallow abstraction. @brauliodiez can you check if it is clearer now?

If not, we can arrange a session (monday maybe) and I'll try to explain it in person. I would need drawings I guess. Or if you prefer, I can try to create a page in the wiki with the basics behind the synchronization. 

[Link to commit](https://github.com/Lemoncode/LeanMood/pull/168/commits/d9cd50cd321fb25e7f3fa0159abe6bb07c3ef661)

Reply:
Let's discuss it on monday, I think the mehtod could be simpflied, e.g. the switch statement can be encapsulated in a function (following the single responsbility principle)